### DISCUSSION TITLE
Article publish date is shown as "January 1, 1"

### DISCUSSION THREAD

**Discussion by FuadEfendi** [razonyang/hugo-theme-bootstrap#692] [Q&A] [answered]
Hello,

The date is not shown; it is with latest master branch; even `hugo-theme-bootstrap-skeleton` run locally without any changes shows `January 1, 1` on the http://localhost:1314/series/guide/ (except "More Image Processing Methods" which shows correct date from remote Git repo)

I thought "date" is taken from GitHub repo (at least, I see this from the configuration structure); but "More Image Processing Methods" has it explicitly as a front matter parameter:
date: 2022-07-08T11:34:47+08:00

Can you please confirm that date should be "commit date" taken from repository? Can I use private repository with Netlify? Thanks

**Comment by razonyang** [comment, marked as answer]
Hi, that's expected, since those posts without `date` parameter in their front matter, I'll handle those sample posts later.

**Reply by FuadEfendi** [reply]
Ok, sorry for misunderstanding... I really thought date will be taken from repository commit timestamp during build. 

I just found one need to explicitly set date as in this sample file:

https://raw.githubusercontent.com/razonyang/hugo-theme-bootstrap/master/exampleSite/content/docs/getting-started/introduction/index.md

Perhaps there are some GitHub hooks with placeholders which can change this date on each commit.

**Reply by razonyang** [reply]
@FuadEfendi Sounds good idea that using Git date if there is no `date` specified, I'll implement it if possible.

**Reply by razonyang** [reply]
I just did some research about this, Hugo provides [Git Info](https://gohugo.io/variables/git/), but there is nothing useful for `date` parameter, furthermore `hugo new` will take care of the `date`, that is, the `date` parameter will be created automatically, there are just mistakes of those sample posts.
So I would leave it as it is at the moment.

**Reply by FuadEfendi** [reply]
Hey Razon Yang,

I did search too because I was feeling this requirement is very natural, and found the answer: just add _frontmatter.yaml_ config file with this content:
`date: ["date", ":git"]`

(since your theme uses "date", ":git" will override "date" if it is not explicitly defined in a page). And it works! Of course we need "enableGitInfo = true" too.

[Last Modified Date in Hugo](https://mertbakir.gitlab.io/hugo/last-modified-date-in-hugo/)

**Reply by razonyang** [reply]
Didn't know that, thanks for mentioning that config.
